cinema/views.py

```diff
@@ -2,0 +1,65 @@
+from rest_framework import viewsets
+
+from cinema.models import Genre, Actor, Movie, MovieSession, CinemaHall
+from cinema.serializers import (
+    GenreSerializer,
+    ActorSerializer,
+    MovieSerializer, MovieSessionSerializer, CinemaHallSerializer,
+    MovieListSerializer, MovieRetrieveSerializer,
+    MovieSessionRetrieveSerializer,
+)
+
+
+class GenreViewSet(viewsets.ModelViewSet):
+    queryset = Genre.objects.all()
+    serializer_class = GenreSerializer
+
+
+class ActorViewSet(viewsets.ModelViewSet):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.all()
+    serializer_class = MovieSerializer
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MovieListSerializer
+        elif self.action == "retrieve":
+            return MovieRetrieveSerializer
+
+        return MovieSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        if self.action in ("list", "retrieve"):
+            return queryset.prefetch_related("genres", "actors")
+        return queryset
```

Comment:
Consider adding type annotations to the `get_queryset` method to improve code readability and maintainability. This is a requirement according to the task checklist.